xdsl/transforms/scf_parallel_loop_tiling.py

```diff
@@ -0,0 +1,147 @@
+                    affine.AffineExpr.dimension(1) - affine.AffineExpr.dimension(2),
+                ),
+            )
+        )
+        for i in range(len(op.lowerBound)):
+            if i in tile_sizes:
+                inner_lower.append(zero)
+                ilower, iupper, istep = lower[i], upper[i], step[i]
+                if (
+                    isinstance(ilower, arith.Constant)
+                    and isinstance(iupper, arith.Constant)
+                    and isinstance(istep, arith.Constant)
+                ):
+                    lower_v, upper_v, step_v = (
+                        c.value for c in (ilower, iupper, istep)
+                    )
+                    assert isa(lower_v, IntegerAttr[IndexType])
+                    assert isa(upper_v, IntegerAttr[IndexType])
+                    assert isa(step_v, IntegerAttr[IndexType])
+                    lower_v, upper_v, step_v = (
+                        c.value.data for c in (lower_v, upper_v, step_v)
+                    )
+                    iter_count = (upper_v - lower_v) // step_v
+                    if iter_count % tile_sizes_v[i] == 0:
+                        inner_upper.append(tile_sizes[i])
+                        continue
+
+                arg_index = tiled_dims.index(i)
+                minop = affine.MinOp(
+                    operands=[
+                        [
+                            tile_sizes[i],
+                            outter_upper[arg_index],
+                            outter_loop.body.block.args[arg_index],
+                        ]
+                    ],
+                    properties={"map": minmap},
+                    result_types=[IndexType()],
+                )
+                minops.append(minop)
+                inner_upper.append(minop)
+
+            else:
+                inner_lower.append(lower[i])
+                inner_upper.append(upper[i])
+
+        inner_loop = ParallelOp(inner_lower, inner_upper, step, op.detach_region(0))
+        for i, arg in reversed(list(enumerate(inner_loop.body.block.args))):
+            if i in tile_sizes:
+                arg_index = tiled_dims.index(i)
+                iv = arith.Addi(outter_loop.body.block.args[arg_index], arg)
+                assert inner_loop.body.block.first_op is not None
+                inner_loop.body.block.insert_op_before(
+                    iv, inner_loop.body.block.first_op
+                )
+                for use in tuple(arg.uses):
+                    if use.operation is iv:
+                        continue
+                    use.operation.operands[use.index] = iv.result
+        outter_loop.body.block.insert_ops_before([*minops, inner_loop], outter_yield)
+        rewriter.replace_matched_op(
+            [zero, *tile_sizes.values(), *outter_step, outter_loop]
+        )
+
+
+@dataclass
+class ScfParallelLoopTilingPass(ModulePass):
+    name = "scf-parallel-loop-tiling"
+
+    parallel_loop_tile_sizes: list[int]
```

Comment:
would this make sense to be a tuple? do we want it to be mutable? similarly above

Reply:
That's our Pass parameters implementation. Using tuple errors out as the infrastructure expects lists; I just tried and just switching to a tuple here yields an error; I'll let you open an issue to change this if you want?

Reply:
ah yes of course my bad

Reply:
but yes I think that's worth an issue

Reply:
#1858

Reply:
How about for the rewrite? Seems like that could be a tuple today

Reply:
Sure